kernel/arch/dreamcast/include/dc/dmac.h

```diff
@@ -0,0 +1,129 @@
+/* KallistiOS ##version##
+
+   dmac.h
+   Copyright (C) 2023 Andy Barajas
+
+*/
+
+/** \file   dc/dmac.h
+    \brief  Macros to access the DMAC(Direct Memory Access Controller) 
+    registers. 
+
+    This header provides a set of macros to facilitate checking
+    the values of various DMA channels on the system.
+
+    DMA channel 0 and its registers (DMAC_SAR0, DMAC_DAR0, DMAC_DMATCR0, 
+    DMAC_CHCR0) are used by the hardware and not accessible to us but are 
+    documented here anyway.
+
+    DMA channel 2 is strictly used to transfer data to the PVR/TA.
+
+    DMA channel 1 & 3 are free to use.
+*/
+
+#ifndef __DC_DMAC_H
+#define __DC_DMAC_H
+
+#include <sys/cdefs.h>
+__BEGIN_DECLS
+
+#define DMAC_BASE     0xffa00000
+
+/** \defgroup dma_sar_registers  DMA Source Address Registers (SAR0-SAR3)
```

Comment:
```suggestion
/** \name DMA Source Address Registers (SAR0-SAR3)
```
Turns out this is much better for grouping things within a header file in Doxygen. Otherwise we're polluting the global topics/groups page with top-level groups floating around for everything.